app/models/manageiq/providers/vmware/infra_manager/inventory/parser/host_system.rb

```diff
@@ -73,7 +73,7 @@ def parse_host_system_host_networks(host, hardware, props)
       vnics = Array(network[:consoleVnic]) + Array(network[:vnic])
       vnics.each do |vnic|
         port_key = vnic.port
-        portgroup = network[:portgroup].find { |pg| pg.port.find { |p| p.key == port_key } }
+        portgroup = Array(network[:portgroup]).find { |pg| pg.port.find { |p| p.key == port_key } }
```

Comment:
Not directly related to reconnect, but I noticed when testing a new host being added this was failing